sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3997,6 +3997,19 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         c.markAsAnalyzed(AnalysisContext.get)
       case c: KeepAnalyzedQuery if c.resolved =>
         c.storeAnalyzedQuery()
+      case c: CreateTable if c.resolved =>
```

Comment:
I went for handling CREATE/REPLACE directly as these two commands convert defaults to V2. There are a few other commands for dealing with variables that also use `DefaultValueExpression`. I didn't handle them here as there is no need to convert their defaults to V2. I can make this rule more generic by calling `transformExpressions` on arbitrary commands and defining a separate node pattern for `DefaultValueExpression`. It will be more generic but complicated. I am not sure we need the extra complexity now.